pagarme/react-style-guide
# Add new lib for destructuring sort

This PR adds new lib for destructuring parameter order validation.

## How to test

1. Clone this project
```
git clone -b update/dependencies git@github.com:pagarme/react-style-guide.git
```

2. Go to the eslint project folder and create a yarn link(this should create a link called `eslint-config-pagarme-react`)
```
  cd react-style-guide/eslint
  yarn link
```

3. Do de same for the stylelint project(this should create a link called `stylelint-config-pagarme-react`)
```
  cd react-style-guide/stylelint
  yarn link
```

4. In [Former kit project](https://github.com/pagarme/former-kit) link the created reference of `eslint-config-pagarme-react` using the following command:
```
  yarn link eslint-config-pagarme-react
```

5. In [Former kit folder](https://github.com/pagarme/former-kit) install all dependencies including the new dependencies related to the added packs.
```
  yarn
```

6.  Run the lint command to test the new rules. This should trigger some errors related to the new rules.
```
  yarn lint
```

This process should create some errors and ask for some packages installation(probably `eslint-plugin-react-hooks` and `eslint-plugin-sort-destructure-keys`). The final result using the [Former kit's](https://github.com/pagarme/former-kit) master branch should create a lot of erros like the following error: 
<img width="1436" alt="Screen Shot 2019-04-24 at 15 22 58" src="https://user-images.githubusercontent.com/1773766/56683888-e722d480-66a4-11e9-9e5a-500573e6cffd.png">
